test/src/code_field/split_highlight_lines_test.dart

```diff
@@ -1,67 +1,150 @@
+import 'package:flutter_code_editor/src/highlight/node.dart';
 import 'package:flutter_code_editor/src/highlight/result.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:highlight/highlight.dart';
 import 'package:highlight/languages/dart.dart';
 
 const examples = {
   //
-  'Empty': '',
+  'Empty': _Example(text: '', expected: '{value: \'\'}'),
 
-  'Single line comment': '''
+  'Single line comment': _Example(
+    text: '''
 public class MyClass {
   public void main() { //comment
   }
 }''',
+    expected: 
+'''
+{value: 'public '}
+{className: class, children: ({value: ''}, {className: keyword, children: ({value: 'class'})}, {value: ' '}, {className: title, children: ({value: 'MyClass'})}, {value: ' '})}
```

Comment:
It's hard to read this.
I wish we had something like this:

```
{value: 'public '}
{className: class}
  {value: ''}
  {className: keyword}
    {value: 'class'}
...
```

Also with this format `value: null` will not be an obstacle.